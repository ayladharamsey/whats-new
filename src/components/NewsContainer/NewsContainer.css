main {
    display: flex;
    flex-wrap: wrap;
    width: 80vw;
    font-family: Roboto;
    overflow: scroll;
    height: 100vh;
    margin-left: 15px;
}

