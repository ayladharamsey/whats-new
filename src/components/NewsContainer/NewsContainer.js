import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

const NewsContainer = props => {
    return (
        <main>
            {props.local.map(newsStory => <NewsArticle key={newsStory.id} news={newsStory}/>)}
        </main>
    )
}

export default NewsContainer;