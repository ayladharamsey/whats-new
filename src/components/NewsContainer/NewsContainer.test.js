import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer'

describe('NewsContainer', () => {
    let wrapper, stories, selected; 

    beforeEach(() => {
        stories = [{description: 'yas queen', headline: 'i rule the school', id: 3, url: 'www.coolkids.com', img: 'www.potatoes.com'}];
        selected = {description: 'yas queen', headline: 'i rule the school', id: 3, url: 'www.coolkids.com', img: 'www.potatoes.com'};
        wrapper = shallow(<NewsContainer newsArticles = {stories} selectedArticle={selected} linkOut={jest.fn()}/>)
    })

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })


});