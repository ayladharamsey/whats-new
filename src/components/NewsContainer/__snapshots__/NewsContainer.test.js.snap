// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<main>
  <NewsArticle
    key="3"
    linkOut={[MockFunction]}
    story={
      Object {
        "description": "yas queen",
        "headline": "i rule the school",
        "id": 3,
        "img": "www.potatoes.com",
        "url": "www.coolkids.com",
      }
    }
  />
</main>
`;
